### PR TITLE
blui 4671 update js docs

### PR DESCRIPTION
**Pull request by JeffGreiner-eaton**
Fixes BLUI-4671 .

#### Changes proposed in this Pull Request:

- add JS doc section for prop types 
- area's marked TODO need assistance
- TODO with all types in context dir

#### Screenshots / Screen Recording (if applicable)

-

#### To Test:

-

#### Any specific feedback you are looking for?

-

### CODE REVIEW

File: login-workflow/src/components/RegistrationWorkflow/RegistrationWorkflow.tsx [hunk 1]
```diff
@@ -23,9 +23,25 @@ import { parseQueryString } from '../../utils';
...
+     */
     initialScreenIndex?: number;
+
+    /**
+     * The success screen to display upon successful registration
+     */
     successScreen?: JSX.Element;
+
+    /**
+     * Boolean when true verifies validateUserRegistrationRequest for verifyCode
```
Comment (joebochill) [inline login-workflow/src/components/RegistrationWorkflow/RegistrationWorkflow.tsx:37]: This description may need a little cleaning up. Should mention that invite registration truncates the default list of screens to skip email entry and validation and pulls the validation code from the url query string instead.

File: login-workflow/src/components/RegistrationWorkflow/RegistrationWorkflow.tsx [hunk 2]
```diff
@@ -23,9 +23,25 @@ import { parseQueryString } from '../../utils';
...
+     * @default false
+     */
     isInviteRegistration?: boolean;
+
+    /**
+     * Component that displays the success screen
```
Comment (joebochill) [inline login-workflow/src/components/RegistrationWorkflow/RegistrationWorkflow.tsx:43]:
```suggestion
     * Component to display for the success screen if the account already exists.
```

File: login-workflow/src/components/SetPassword/types.ts [hunk 3]
```diff
@@ -4,19 +4,73 @@ import { MutableRefObject } from 'react';
...
  */
 export type PasswordRequirement = {
+    /**
+     * The description of the password requirement
+     */
     description: string;
+
+    /**
+     * The regex to validate the password
+     * @returns RegExp
```
Comment (joebochill) [inline login-workflow/src/components/SetPassword/types.ts:14]: Not a function, so no returned value
```suggestion
```

File: login-workflow/src/components/SetPassword/types.ts [hunk 4]
```diff
@@ -4,19 +4,73 @@ import { MutableRefObject } from 'react';
...
+    /**
+     * Function called when the new password or confirm new password fields value changes
+     * @param {string} password - new password value
+     * @param {string} confirm - confirm password value
+     * @returns void
+     */
     onPasswordChange?: (passwords: { password: string; confirm: string }) => void;
+
+    /**
+     * The label for the new password field (default = 'Password')
```
Comment (joebochill) [inline login-workflow/src/components/SetPassword/types.ts:29]: For the defaults in this file, use the @default keyword instead

File: login-workflow/src/components/SetPassword/types.ts [hunk 5]
```diff
@@ -4,19 +4,73 @@ import { MutableRefObject } from 'react';
...
     confirmPasswordLabel?: string;
+
+    /**
+     * The initial value for the confirm password
+     */
     initialConfirmPasswordValue?: string;
+
+    /**
+     * The requirements to set password
+     * TODO anything for array?
```
Comment (joebochill) [inline login-workflow/src/components/SetPassword/types.ts:50]:
```suggestion
     * A list of password complexity requirements
```

File: login-workflow/src/components/SetPassword/types.ts [hunk 6]
```diff
@@ -4,19 +4,73 @@ import { MutableRefObject } from 'react';
...
+     */
     passwordRequirements?: PasswordRequirement[];
+
+    /**
+     * The ref to forward to the password input
+     * TODO for object
```
Comment (joebochill) [inline login-workflow/src/components/SetPassword/types.ts:56]: Nothing needed here
```suggestion
```

File: login-workflow/src/components/SetPassword/types.ts [hunk 7]
```diff
@@ -4,19 +4,73 @@ import { MutableRefObject } from 'react';
...
+     */
     passwordRef?: MutableRefObject<any>;
+
+    /**
+     * The ref to forward to the confirm password input
+     * TODO for object
```
Comment (joebochill) [inline login-workflow/src/components/SetPassword/types.ts:62]:
```suggestion
```

File: login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts [hunk 8]
```diff
@@ -2,36 +2,108 @@ import { CardActionsProps, CardHeaderProps, TypographyProps } from '@mui/materia
...
 export type ErrorStateProps = TypographyProps & {
+    /**
+     * The text to display for error
+     */
     message?: string;
 };
 
 export type WorkflowCardBaseProps = BoxProps & {
+    /**
+     * Boolean value for isLoading
```
Comment (joebochill) [inline login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts:13]:
```suggestion
     * If true, a blocking progress spinner will be displayed over the card
```

File: login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts [hunk 9]
```diff
@@ -2,36 +2,108 @@ import { CardActionsProps, CardHeaderProps, TypographyProps } from '@mui/materia
...
+     */
     loading?: boolean;
-    backgroundImage?: string; // card background
-    error?: boolean | string; // each screen should have an error state
+
+    /**
+     * To display card background
```
Comment (joebochill) [inline login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts:18]:
```suggestion
     * A custom background to render behind the card
```

File: login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts [hunk 10]
```diff
@@ -2,36 +2,108 @@ import { CardActionsProps, CardHeaderProps, TypographyProps } from '@mui/materia
...
+     */
+    backgroundImage?: string;
+
+    /**
+     * The error state for each workflowCard
```
Comment (joebochill) [inline login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts:23]:
```suggestion
     * An error message to render for the card
```

File: login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts [hunk 11]
```diff
@@ -2,36 +2,108 @@ import { CardActionsProps, CardHeaderProps, TypographyProps } from '@mui/materia
...
+     * Whether or not to show a divider below the instructions
+     */
     divider?: boolean;
 };
 
 // type DataObject = { [key: string]: any };
 
 export type WorkflowCardActionsProps = CardActionsProps & {
+    /**
+     * Boolean value to display a divider above workflow card action buttons
```
Comment (joebochill) [inline login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts:46]:
```suggestion
     * True to display a divider above workflow card action buttons
```

File: login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts [hunk 12]
```diff
@@ -2,36 +2,108 @@ import { CardActionsProps, CardHeaderProps, TypographyProps } from '@mui/materia
...
     previousLabel?: string;
+
+    /**
+     * The label to display for the next button
+     */
     nextLabel?: string;
+
+    /**
+     * Function called when the previous button is clicked
+     * @param {string} key TODO here data?
```
Comment (joebochill) [inline login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts:82]:
```suggestion
     * @param {Object} data - data collected from the card / screen
```

File: login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts [hunk 13]
```diff
@@ -2,36 +2,108 @@ import { CardActionsProps, CardHeaderProps, TypographyProps } from '@mui/materia
...
+     * @returns void
+     */
     onPrevious?: (data?: { [key: string]: any }) => void;
+
+    /**
+     * Function that is called when the next button is clicked
+     * @param {string} key TODO here data?
+     * @returns void
+     */
```
Comment (joebochill) [inline login-workflow/src/components/WorkflowCard/WorkflowCard.types.ts:91]:
```suggestion
    /**
     * Function that is called when the next button is clicked
     * @param {Object} data - data collected from the card / screen
     * @returns void
     */
```

File: login-workflow/src/contexts/RegistrationContext/types.ts [hunk 14]
```diff
@@ -11,12 +11,54 @@ export type AccountDetails = {
...
+     * A function that is used to validate a registration code. This function will be called when the user clicks the Next button on the Verify Code screen screen
+     * @param {string} validationCode - verify code input
+     * @param {string} validationEmail - verify email input
+     * @returns Promise<boolean>
+     */
     validateUserRegistrationRequest?: (validationCode: string, validationEmail?: string) => Promise<boolean>;
+
+    /**
+     * A function that is used to create a password. This function will be called when the user clicks the Next button on the Create Password screen
+     * @param {string} password - validates password input
```
Comment (joebochill) [inline login-workflow/src/contexts/RegistrationContext/types.ts:44]:
```suggestion
     * @param {string} password - the provided password
```

File: login-workflow/src/contexts/RegistrationContext/types.ts [hunk 15]
```diff
@@ -11,12 +11,54 @@ export type AccountDetails = {
...
+    /**
+     * A function that is used to request a registration code.
+     * @param {string} email - This function will be called when the user lands on the Verify Code screen as well as when a user clicks the Resend Verification Code button
+     * @returns Promise<string>
+     */
     requestRegistrationCode?: (email: string) => Promise<string>;
+
+    /**
+     * A function that is used to validate a registration code. This function will be called when the user clicks the Next button on the Verify Code screen screen
+     * @param {string} validationCode - verify code input
```
Comment (joebochill) [inline login-workflow/src/contexts/RegistrationContext/types.ts:36]:
```suggestion
     * @param {string} validationCode - the provided verification code
```

File: login-workflow/src/contexts/RegistrationContext/types.ts [hunk 16]
```diff
@@ -11,12 +11,54 @@ export type AccountDetails = {
...
+     * @param {string} validationEmail - verify email input
```
Comment (joebochill) [inline login-workflow/src/contexts/RegistrationContext/types.ts:37]:
```suggestion
     * @param {string} validationEmail - the provided email address
```

File: login-workflow/src/contexts/RegistrationContext/types.ts [hunk 17]
```diff
@@ -11,12 +11,54 @@ export type AccountDetails = {
...
+    /**
+     * A function that is used to create a password. This function will be called when the user clicks the Next button on the Create Password screen
+     * @param {string} password - validates password input
+     * @returns Promise<boolean>
+     */
     createPassword?: (password: string) => Promise<boolean>;
+
+    /**
+     * A function that is used to set the account details. This function will be called when the user clicks the Next button on the Account Details screen
+     * @param {AccountDetails} details - verify first and last name input
```
Comment (joebochill) [inline login-workflow/src/contexts/RegistrationContext/types.ts:51]:
```suggestion
     * @param {AccountDetails} details - information collected from the screen, including firstName and lastName
```

File: login-workflow/src/contexts/RegistrationContext/types.ts [hunk 18]
```diff
@@ -11,12 +11,54 @@ export type AccountDetails = {
...
+     * @returns Promise<boolean>
+     */
     setAccountDetails?: (details: AccountDetails) => Promise<boolean>;
+
+    /**
+     * A function that is used to complete the registration. This function will be called when the user clicks the Next button on the final registration workflow screen
+     * @param {string} email - Used to display on final screen
+     * @param {string} organizationName - Used to display on final screen
```
Comment (joebochill) [inline login-workflow/src/contexts/RegistrationContext/types.ts:59]: These are not the params here...the params are userData and validationCode. email and organizationName are part of the @returns value...if you want to describe what they are used for you can do it at the end of the @returns line

File: login-workflow/src/contexts/RegistrationContext/types.ts [hunk 19]
```diff
@@ -25,10 +67,39 @@ export type RegistrationUIActions = {
 };
 
 export type RegistrationContextProviderProps = {
+    /**
+     * An object of functions that are used to manage the authentication workflow
+     * @object RegistrationUIActions
```
Comment (joebochill) [inline login-workflow/src/contexts/RegistrationContext/types.ts:72]: This type is changing, but as written it should be
```suggestion
     * @returns RegistrationUIActions
```

File: login-workflow/src/contexts/RegistrationContext/types.ts [hunk 20]
```diff
@@ -25,10 +67,39 @@ export type RegistrationUIActions = {
...
+    /**
+     * A function that is used to navigate to a new URL. This is used to navigate to the various screens of the workflow
+     * @param {string} url - path to url
+     * @returns void
+     */
     navigate: (url: string) => void;
+
+    /**
+     * An object that defines the various routes for the workflow
+     * @object RouteConfig
```
Comment (joebochill) [inline login-workflow/src/contexts/RegistrationContext/types.ts:90]:
```suggestion
```

File: login-workflow/src/contexts/RegistrationContext/types.ts [hunk 21]
```diff
@@ -25,10 +67,39 @@ export type RegistrationUIActions = {
...
+     */
     routeConfig: RouteConfig;
+
+    /**
+     * An optional i18n object that is used to translate the UI. This is only needed if you want to use custom translation keys / languages inside any of the workflow screens
+     * @object i18n
```
Comment (joebochill) [inline login-workflow/src/contexts/RegistrationContext/types.ts:96]:
```suggestion
```

File: login-workflow/src/contexts/RegistrationContext/types.ts [hunk 22]
```diff
@@ -25,10 +67,39 @@ export type RegistrationUIActions = {
...
+     */
     i18n?: i18n; // add languages / override strings in bulk
+
+    /**
+     * An object that is used to configure error handling within the workflow.
+     * @object ErrorContextProviderProps
```
Comment (joebochill) [inline login-workflow/src/contexts/RegistrationContext/types.ts:102]:
```suggestion
```

File: login-workflow/src/screens/EulaScreen/types.ts [hunk 23]
```diff
@@ -3,27 +3,48 @@ import { WorkflowCardProps } from '../../components/WorkflowCard/WorkflowCard.ty
...
-    // EULA checkbox props
+    /**
+     * Used to set checkbox props
+     */
     checkboxProps?: CheckboxProps;
 
-    // true if the EULA should be rendered as HTML
+    /**
+     * The EULA should be rendered as HTML
+     * @default true
```
Comment (joebochill) [inline login-workflow/src/screens/EulaScreen/types.ts:23]: I think the default here is false, no?

File: login-workflow/src/screens/ForgotPasswordScreen/types.ts [hunk 24]
```diff
@@ -54,9 +35,43 @@ export type ForgotPasswordScreenProps = WorkflowCardProps & {
...
+    emailLabel?: string;
+
+    /**
+     * The initial value for the email text field
+     */
+    initialEmailValue?: string;
+
+    /**
+     * @param emailValidator The function used to test the input for valid formatting
+     * @returns boolean | string
```
Comment (joebochill) [inline login-workflow/src/screens/ForgotPasswordScreen/types.ts:50]:
```suggestion
     * A function used to test the input for valid formatting
     * @param {string} email - the provided email address
     * @returns boolean | string
```

File: login-workflow/src/screens/LoginScreen/types.ts [hunk 25]
```diff
@@ -4,44 +4,165 @@ import { ErrorManagerProps } from '../../components/Error';
...
+     * See [MUI's TextFieldProps API](https://mui.com/material-ui/api/text-field/) for more details.
+     */
     usernameTextFieldProps?: TextFieldProps;
+
+    /**
+     * The function used to validate the username
+     * @param {string} foo - validates username
+     * @returns boolean | string
+     */
     usernameValidator?: (foo: string) => boolean | string;
```
Comment (joebochill) [inline login-workflow/src/screens/LoginScreen/types.ts:23]: rename 'foo' to what the variable actually is please...in this case: 'username'

File: login-workflow/src/screens/LoginScreen/types.ts [hunk 26]
```diff
@@ -4,44 +4,165 @@ import { ErrorManagerProps } from '../../components/Error';
...
+
+    /**
+     * whether or not the 'remember me' checkbox should be checked by default
+     * @default false
+     */
     rememberMeInitialValue?: boolean;
+
+    /**
+     * The callback function that is called when the 'remember me' checkbox is changed
+     * @returns void
```
Comment (joebochill) [inline login-workflow/src/screens/LoginScreen/types.ts:68]: missing @param

**Review by surajeaton — COMMENTED** [review]
Please resolve the conflicts.

**Review by daileytj — APPROVED** [review]

**Pull request merged**